File: examples/kotlin/src/main/kotlin/Move.kt
package battlebot

import com.google.gson.Gson

enum class Hand {
    ROCK,
    PAPER,
    SCISSORS
}

data class Move(val move: Hand, val art: List<String> = emptyList()) {
    fun toJSON(): String {
        return Gson().toJson(this)
    }
}

File: examples/swift/Sources/battlebot/move.swift
import Foundation

enum Hand : String, CaseIterable, Codable {
    case ROCK
    case PAPER
    case SCISSORS
}

struct Move : Codable {
    var move : Hand
    var art = ""

    func toJSON() throws -> String {
        let data = try JSONEncoder().encode(self)
        return String(data: data, encoding: .utf8)!
    }
}


